app/core/Snaps/SnapsMethodMiddleware.ts

```diff
@@ -4,16 +4,25 @@ import {
   RequestedPermissions,
   SubjectType,
 } from '@metamask/permission-controller';
+import { SnapRpcHookArgs } from '@metamask/snaps-utils';
 import { RestrictedMethods } from '../Permissions/constants';
+import { keyringSnapPermissionsBuilder } from '../SnapKeyring/keyringSnapsPermissions';
+import { SnapId } from '@metamask/snaps-sdk';
+import { EngineContext } from '../Engine';
+import { handleSnapRequest } from './utils';
 
+export function getSnapIdFromRequest(
+  request: Record<string, unknown>,
+): string | null {
+  const snapId = request.snapId;
```

Comment:
What about de-structuring here:

```suggestion
  const { snapId } = request;
```

Reply:
clean